util/carvelhelpers/package.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package carvelhelpers
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	yttui "github.com/vmware-tanzu/carvel-ytt/pkg/cmd/ui"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/files"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/workspace"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/workspace/datavalues"
+)
+
+const ConfigFilePermissions = 0o600
+
+// ProcessYTTPackage processes configuration directory with ytt tool
+// Implements similar functionality as `ytt -f <config-dir>`
+func ProcessYTTPackage(configDirs ...string) ([]byte, error) {
+	yttFiles, err := files.NewSortedFilesFromPaths(configDirs, files.SymlinkAllowOpts{})
+	if err != nil {
+		return nil, err
+	}
+
+	lib := workspace.NewRootLibrary(yttFiles)
+	libCtx := workspace.LibraryExecutionContext{Current: lib, Root: lib}
+	libExecFact := workspace.NewLibraryExecutionFactory(&NoopUI{}, workspace.TemplateLoaderOpts{})
+	loader := libExecFact.New(libCtx)
+
+	valuesDoc, libraryValueDoc, err := loader.Values([]*datavalues.Envelope{}, datavalues.NewNullSchema())
+	if err != nil {
+		return nil, err
+	}
+	result, err := loader.Eval(valuesDoc, libraryValueDoc, []*datavalues.SchemaEnvelope{})
+	if err != nil {
+		return nil, err
+	}
+	return result.DocSet.AsBytes()
+}
+
+// CarvelPackageProcessor processes a carvel package and returns a configuration YAML file
+func CarvelPackageProcessor(pkgDir string) ([]byte, error) {
+	// Each package contains `config` and `.imgpkg` directory
+	// `config` directory contains ytt files
+	// `.imgpkg` directory contains ImageLock configuration for ImageResolution
+	configDir := filepath.Join(pkgDir, "config")
+	file := []string{configDir}
+	bytes, err := ProcessYTTPackage(file...)
+	if err != nil {
+		return nil, errors.Wrap(err, "error while running ytt")
+	}
+
+	f, err := os.CreateTemp("", "ytt-processed")
+	if err != nil {
+		return nil, errors.Wrap(err, "error while creating temp directory")
+	}
+	defer os.Remove(f.Name())
+	err = SaveFile(f.Name(), bytes)
+	if err != nil {
+		return nil, errors.Wrap(err, "error while saving file")
+	}
+
+	inputFilesForImageResolution := []string{f.Name()}
+
+	// Use `.imgpkg` directory if exists for ImageResolution
+	imgpkgDir := filepath.Join(pkgDir, ".imgpkg")
+	if PathExists(imgpkgDir) {
+		inputFilesForImageResolution = append(inputFilesForImageResolution, imgpkgDir)
+	}
+	return ResolveImagesInPackage(inputFilesForImageResolution)
+}
+
+// SaveFile saves the file to the provided path
+// Also creates missing directories if any
+func SaveFile(filePath string, data []byte) error {
+	dirName := filepath.Dir(filePath)
+	if _, serr := os.Stat(dirName); serr != nil {
```

Comment:
I can't tell, but looks like the `serr` information doesn't get passed along. Do you think that's all right? Maybe people don't need to know what `serr` is because you pass along `merr` later? Kind of strange, but I guess it can  work.